### PR TITLE
feat: [PPR 458] definizione api del microservizio micronaut

### PR DESCRIPTION
**Pull request by cap-ang**
#### List of Changes

- Swagger and Swagger UI configuration with OpenAPI Specification
- REST Base API for health check monitoring definition
- Model class for OpenAPI Spec schema definition
- Redirect to swagger-ui for GET request mapping to REST root resource '/'
- REST API definition for organization resource

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,6 +31,16 @@
     </repositories>
 
     <dependencies>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.12</version>
```
Comment (pasqualespica) [inline pom.xml]: create new system properties ex. `<lombok.version>`

File: pom.xml [hunk 2]
```diff
@@ -98,15 +108,37 @@
                     <!-- <useIncrementalCompilation>false</useIncrementalCompilation> -->
 
                     <annotationProcessorPaths combine.children="append">
+                        <path>
+                            <!-- must precede micronaut-inject-java -->
```
Comment (pasqualespica) [inline pom.xml:114]: why ? What's the problem ?

File: pom.xml [hunk 3]
```diff
@@ -98,15 +108,37 @@
...
+                        </path>
                         <path>
                             <groupId>io.micronaut</groupId>
                             <artifactId>micronaut-http-validation</artifactId>
                             <version>${micronaut.version}</version>
                         </path>
+                        <path>
+                            <groupId>io.micronaut.openapi</groupId>
+                            <artifactId>micronaut-openapi</artifactId>
+                            <version>4.5.2</version>
```
Comment (pasqualespica) [inline pom.xml]: ditto ( new sys pom prop )

File: src/main/java/it/gov/pagopa/reportingorgsenrollement/Application.java [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
 
 import io.micronaut.runtime.Micronaut;
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.info.Contact;
+import io.swagger.v3.oas.annotations.info.Info;
+import io.swagger.v3.oas.annotations.info.License;
 
+@OpenAPIDefinition(
+        info = @Info(
+                title = "Reporting Organizations Enrollments",
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/reportingorgsenrollement/Application.java]: lift & shift this info into [configuration file](https://micronaut-projects.github.io/micronaut-openapi/latest/guide/#propertiesFileConfiguration)

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (cap-ang) [reply pom.xml:114]: This order guarantees that the Lombok processor will run before the Micronaut processor.
The issue is documented among the micronaut-core issues ([here](https://github.com/micronaut-projects/micronaut-core/issues/218)).

**Review by pasqualespica — APPROVED** [review]
👍

**Review by pasqualespica — APPROVED** [review]
great

**Pull request merged**